### PR TITLE
Run GPU e2e tests on A3+ cluster

### PR DESCRIPTION
**Pull request by michelle-yooh**
# Description

A3+ XLML tests

# Tests

Please describe the tests that you ran on Cloud VM to verify changes.

**Instruction and/or command lines to reproduce your tests:** ...

**List links for your tests (use go/shortn-gen for any internal link):** ...

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [X] I have performed a self-review of my code.
- [X] I have necessary comments in my code, particularly in hard-to-understand areas.
- [X] I have run one-shot tests and provided workload links above if applicable. 
- [X] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

File: dags/multipod/maxtext_end_to_end.py [hunk 1]
```diff
@@ -158,7 +158,30 @@
...
+    ).run()
+    stable_a3plus_gpu = gke_config.get_maxtext_end_to_end_gpu_gke_test_config(
+        accelerator_type=GpuVersion.XPK_H100_MEGA,
+        gpu_zone=Zone.US_EAST4_A.value,
+        time_out_in_min=300,
+        test_name=f"{test_name_prefix}-stable-{model}",
+        run_model_cmds=(test_script,),
+        num_slices=nnodes,
+        cluster_name=ClusterName.A3PLUS_CLUSTER.value,
+        docker_image="gcr.io/supercomputer-testing/yooh/maxtext-tcpx-stable",
```
Comment (yangyuwei) [inline dags/multipod/maxtext_end_to_end.py:180]: Ditto. Can we use DockerImage.MAXTEXT_GPU_JAX_STABLE.value?

Comment (yangyuwei) [inline dags/multipod/maxtext_end_to_end.py:180]: Can we use DockerImage.MAXTEXT_GPU_JAX_STABLE.value?

File: dags/multipod/maxtext_end_to_end.py [hunk 2]
```diff
@@ -158,7 +158,30 @@
...
-    pinned_gpu >> stable_gpu
+    pinned_a3plus_gpu = gke_config.get_maxtext_end_to_end_gpu_gke_test_config(
+        accelerator_type=GpuVersion.XPK_H100_MEGA,
+        gpu_zone=Zone.US_EAST4_A.value,
+        time_out_in_min=300,
+        test_name=f"{test_name_prefix}-pinned-{model}",
+        run_model_cmds=(test_script,),
+        num_slices=nnodes,
+        cluster_name=ClusterName.A3PLUS_CLUSTER.value,
+        docker_image="gcr.io/supercomputer-testing/yooh/maxtext-tcpx",
```
Comment (yangyuwei) [inline dags/multipod/maxtext_end_to_end.py:169]: Can we use DockerImage.MAXTEXT_GPU_JAX_PINNED.value?

File: dags/multipod/maxtext_end_to_end.py [hunk 3]
```diff
@@ -160,7 +160,7 @@
     ).run()
     pinned_a3plus_gpu = gke_config.get_maxtext_end_to_end_gpu_gke_test_config(
         accelerator_type=GpuVersion.XPK_H100_MEGA,
-        gpu_zone=Zone.US_EAST4_A.value,
+        gpu_zone=Zone.US_EAST1_C.value,
```
Comment (yangyuwei) [inline dags/multipod/maxtext_end_to_end.py:163]: The newly created `a3plus-benchmark` cluster is in US_CENTRAL1_C.

File: dags/multipod/maxtext_end_to_end.py (hunk 1 above)
Comment (michelle-yooh) [reply dags/multipod/maxtext_end_to_end.py]: yes just pushed this commit, thanks for pointing out

**Review by NinaCai — APPROVED** [review]

File: dags/multipod/maxtext_end_to_end.py (hunk 3 above)
Comment (michelle-yooh) [reply dags/multipod/maxtext_end_to_end.py]: addressed in https://github.com/GoogleCloudPlatform/ml-auto-solutions/pull/313/commits/db478c7f111aeec553696a40e619edc09b5b6f72

**Review by jonb377 — APPROVED** [review]

File: dags/multipod/maxtext_end_to_end.py (hunk 2 above)
Comment (michelle-yooh) [reply dags/multipod/maxtext_end_to_end.py]: addressed in https://github.com/GoogleCloudPlatform/ml-auto-solutions/commit/db478c7f111aeec553696a40e619edc09b5b6f72

Comment (michelle-yooh) [reply dags/multipod/maxtext_end_to_end.py]: actually this is the correct commit number addressing this change: https://github.com/GoogleCloudPlatform/ml-auto-solutions/pull/313/commits/735b4684347b4bb65ad0dafec2d71e0cd2382ab6

**Review by yangyuwei — APPROVED** [review]

**Review by RissyRan — APPROVED** [review]

**Pull request merged**